components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -4367,4 +4368,69 @@ public boolean isClaimReferredByAnyIdp(Connection dbConnection, String claimUri,
             }
         }
     }
+
+    /**
+     * Method that retrieves identityProvider details of a idpId list.
+     *
+     * @param dbConnection Optional DB connection.
+     * @param tenantId Tenant id.
+     * @param idpIds Set of identity provider ids.
+     * @return A list of identity providers.
+     * @throws IdentityProviderManagementException
+     */
+    public List<IdentityProvider> getIdPsById(Connection dbConnection, int tenantId, Set<String> idpIds)
```

Comment:
1. Let's modify this method to get only IdpNames.
2. Let's remove connection parameter